### PR TITLE
feat: [SRM-13588]: implementing all remaining scenarios for templates and common health source design.

### PR DESCRIPTION
**Pull request by harshilbits**
### Summary
1. Changes in the UI after the backend changes of fields in monitored service .
2. Fixes needed for templates while copying data template fields to verify step.
3. Fixes for validations in templates for verify step.
4. Fixes for validations in templates for Run time screen of verify step.
5. Fixed the connector issue where connector was not getting populated when clicked fields were populated in verify step.
6. Fixed the connector issue where connector was not getting populated for Run time screen of verify step.
7. Making the logic of templates centralised for 
-    Monitored service template.
-    Verify step template 
-    Run time screen of Verify step

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by harshilbits** [comment]
retrigger titlecheck

**Comment by harshilbits** [comment]
retrigger build

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationInputSetStep/components/TemplatisedRunTimeMonitoredService/TemplatisedRunTimeMonitoredService.tsx [hunk 1]
```diff
@@ -25,14 +25,18 @@ import {
...
-import type { ConnectorInfoDTO } from 'services/cv'
-import { healthSourceTypeMapping } from '@cv/pages/monitored-service/MonitoredServiceInputSetsTemplate/MonitoredServiceInputSetsTemplate.utils'
+import type { UpdatedHealthSourceWithAllSpecs } from '@cv/pages/health-source/types'
 import { getMultiTypeInputProps } from '../../../ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/VerificationJobFields/VerificationJobFields.utils'
 import { getRunTimeInputsFromHealthSource } from './TemplatisedRunTimeMonitoredService.utils'
-import { INDEXES } from '../../../ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/SelectMonitoredServiceType/components/MonitoredServiceInputTemplatesHealthSources/MonitoredServiceInputTemplatesHealthSources.constants'
+import {
+  CONNECTOR_REF,
+  INDEXES
+} from '../../../ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/SelectMonitoredServiceType/components/MonitoredServiceInputTemplatesHealthSources/MonitoredServiceInputTemplatesHealthSources.constants'
```
Comment (deepesh-ui) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationInputSetStep/components/TemplatisedRunTimeMonitoredService/TemplatisedRunTimeMonitoredService.tsx:39]: shall we use aliased paths ?

Comment (harshilbits) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationInputSetStep/components/TemplatisedRunTimeMonitoredService/TemplatisedRunTimeMonitoredService.tsx:39]: lint check usually complains if it exceeds the limit and suggest us to use alias path.

**Review by praneshg239 — APPROVED** [review]

**Comment by harshilbits** [comment]
retrigger jest

**Review by deepesh-ui — APPROVED** [review]

**Pull request merged**